sim/kubernetes/engine.go

```diff
@@ -244,7 +244,7 @@ func (kd *kubeEngine) CreateDeployment() (watch.Interface, error) {
 		_, err = kd.client.AppsV1().Deployments(kd.namespace).Create(deployment)
 		if err != nil {
 			fmt.Fprintln(kd.writer, goterm.ResetLine("Creating deployment... failure"))
-			return nil, err
+			return nil, xerrors.Errorf("Deployment creation failed: %w", err)
```

Comment:
I see you wrap all errors, is there a necessity to do that ? I.e the other side is expected to unwrap them for some custom error handling ?
Otherwise by default we normally don't wrap them and use `%v`. Like so we are not committed to the error format we return.

Reply:
Good point. I didn't think enough about this and you're absolutely right, wrapping is a bad idea in nearly all cases here.

Reply:
I've made it uniform across the code base, as there are really no checks on the error type even in dependent projects.